draft-ietf-rtcweb-jsep.xml

```diff
@@ -1568,6 +1584,28 @@ Suggested:
           remote candidate, connectivity checks will be sent to the new
           candidate.</t>
         </section>
+        <section anchor="sec.onicecandidate" numbered="true" toc="default">
+          <name>onicecandidate Event</name>
+          <t>The onicecandidate event is dispatched to the application when the
+          ICE agent has discovered a new allowed local ICE candidate during ICE
+          gathering, as outlined in
+          <xref target="sec.ice-gather-overview" format="default"></xref> and 
+          subject to the restrictions discussed in 
+          <xref target="sec.ice-candidate-policy" format="default"></xref>.
+          The new candidate is supplied as an IceCandidate object 
+          (<xref target="sec.ice-candidate-format" format="default"/>) in 
+          the event, and all of its fields <bcp14>MUST</bcp14> be filled in. 
+          This candidate will also be added to the current and/or pending local
+          description according to the rules defined for Trickle ICE.</t>
+          <t>The onicecandidate event will also be dispatched to indicate
+          end-of-candidates conditions, as defined in 
+          <xref target="RFC8838" format="default"/>. For these indications,
+          the IceCandidate's candidate field <bcp14>MUST</bcp14> be omitted.
```

Comment:
```suggestion
          the IceCandidate's candidate field <bcp14>MUST</bcp14> be an empty string.
```

"This carries the candidate-attribute as defined in section 15.1 of [ICE]. If this RTCIceCandidate represents an end-of-candidates indication or a peer reflexive remote candidate, candidate is an empty string. "

Reply:
The sec.ice-candidate-format xref should cover this, although perhaps it needs to be clarified that that xref applies to this paragraph (about eoc) as well.

Reply:
I'm just being pedantic about "omitted" here. For JS objects that are stringy, there are three "nothing here" values:

```undefined``` [doesn't exist]
```null``` explicitly set to ```null``
```""``` explicitly set to empty string

So, what does "omitted" mean? ```undefined``` is (I think) forbidden by ```DOMString?```, but
I would usually think it meant ```null```.

Reply:
yeah, ```null``` is what we want here. Need to see how I can make this work with the existing text.